force-app/main/default/classes/CRLP_RollupProcessor.cls

```diff
@@ -80,10 +80,17 @@ public inherited sharing class CRLP_RollupProcessor {
     */
     private Id lastParentIdProcessed;
 
+    /**
+    * @description used to flag a condition that requires us NOT to remove lastParentIdProcessed defs.
+    */
+    @TestVisible
+    private Boolean hasNoDetailsForLastParent = false;
```

Comment:
I generally find that variable names are most clear if we can keep them in the positive instead of using negation language so that you don't run into needing to understand double negatives. Could we change this to lastParentHasDetails and then use that in the line 442 if block instead of the negated variable name?